presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1622,7 +1622,14 @@ private static void deleteTemporaryTableDirectories(List<DeclaredIntentionToWrit
     {
         for (DeclaredIntentionToWrite declaredIntentionToWrite : declaredIntentionsToWrite) {
             if (declaredIntentionToWrite.isTemporaryTable()) {
-                deleteRecursivelyIfExists(declaredIntentionToWrite.getContext(), hdfsEnvironment, declaredIntentionToWrite.getStagingPathRoot());
+                String tableName = declaredIntentionToWrite.getSchemaTableName().getTableName();
+                Path tablePath = declaredIntentionToWrite.getStagingPathRoot();
+                if (!tablePath.getName().equals(tableName) &&
```

Comment:
Since we already known is temporary table, what about the following:

```java
    // table path is set to _presto_temporary_table_${uuid}/${uuid}/ for temporary table
    Path tablePath = declaredIntentionToWrite.getStagingPathRoot().getParent();
    checkState(tablePath.getName().equals(tableName);
```

Reply:
Agree

Reply:
Done.